File: Backend/edufun/src/main/java/org/app/edufun/entity/Submission.java
package org.app.edufun.entity;

import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.*;
import org.app.edufun.entity.common.InstantSerializer;
import org.app.edufun.entity.common.SecondsMillisSerializer;

import javax.persistence.*;
import java.sql.Time;
import java.time.Instant;

@Entity
@Table(name="submission")
@AllArgsConstructor
@NoArgsConstructor
@Builder(toBuilder = true)
@Getter
@Setter
@ToString
public class Submission {
    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long submission_id;

    @Column(nullable = false)
    private Long user_id;

    @Column(nullable = false)
    private Long question_id;

    @Column(nullable = false)
    private String submissionTime;

    @JsonSerialize(using = InstantSerializer.class)
    @Column(nullable = false)
    private Instant dt;

    @Column(nullable = false)
    private Integer status;

}

File: Frontend/edu_n_fun/lib/Screens/QuizModule/QuizApiLink/Submission.dart
class Submission {
  final int? submissionId;
  final int questionId;
  final int? userId;
  final String submissionTime;
  final String? dt;
  final int status;

  Submission({
    this.submissionId,
    required this.questionId,
    this.userId,
    required this.submissionTime,
    this.dt,
    required this.status,
  });

  Map<String, dynamic> toJson() {
    return {
      'submissionId': submissionId,  // Corrected field name
      'questionId': questionId,      // Corrected field name
      'userId': userId,              // Corrected field name
      'submissionTime': submissionTime,
      'dt': dt,
      'status': status,
    };
  }

  factory Submission.fromJson(Map<String, dynamic> json) {
    return Submission(
      submissionId: json['submissionId'],
      questionId: json['questionId'],
      userId: json['userId'],
      submissionTime: json['submissionTime'],
      dt: json['dt'],
      status: json['status'],
    );
  }
}